src/flow-manager.c

```diff
@@ -248,6 +254,51 @@ static int FlowManagerFlowTimeout(const Flow *f, enum FlowState state, struct ti
 
     /* do the timeout check */
     if (flow_times_out_at >= ts->tv_sec) {
+        /* for bypassed flow we need to check if the capture method has seen
+         * something and update the logic */
+        FlowBypassInfo *fc = FlowGetStorageById(f, GetFlowBypassInfoID());
+        if (fc) {
+            if (fc->BypassUpdate) {
+                /* flow will be possibly updated */
+                uint64_t pkts_tosrc = fc->tosrcpktcnt;
+                uint64_t bytes_tosrc = fc->tosrcbytecnt;
+                uint64_t pkts_todst = fc->todstpktcnt;
+                uint64_t bytes_todst = fc->todstbytecnt;
+                bool update = fc->BypassUpdate(f, fc->bypass_data, ts->tv_sec);
+                if (update) {
+                    SCLogDebug("Updated flow: %ld", FlowGetId(f));
+                    flow_times_out_at = (int32_t)(f->lastts.tv_sec + timeout);
+                    if (*next_ts == 0 || flow_times_out_at < *next_ts)
+                        *next_ts = flow_times_out_at;
+                    pkts_tosrc = fc->tosrcpktcnt - pkts_tosrc;
+                    bytes_tosrc = fc->tosrcbytecnt - bytes_tosrc;
+                    pkts_todst = fc->todstpktcnt - pkts_todst;
+                    bytes_todst = fc->todstbytecnt - bytes_todst;
+                    if (f->livedev) {
+                        SC_ATOMIC_ADD(f->livedev->bypassed,
+                                      pkts_tosrc + pkts_todst);
+                    }
+                    if (counters) {
+                        counters->bypassed_pkts += pkts_tosrc + pkts_todst;
+                        counters->bypassed_bytes += bytes_tosrc + bytes_todst;
+                    }
+                    return 0;
+                } else {
+                    SCLogDebug("No new packet, dead flow %ld", FlowGetId(f));
+                    if (f->livedev) {
+                        if (FLOW_IS_IPV4(f)) {
+                            LiveDevAddBypassStats(f->livedev, -1, AF_INET);
```

Comment:
this -1 doesn't seem to be handled? I guess the best case is that it adds it to the counter.